src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -2805,7 +2805,13 @@ private void PublishLocally()
         {
             var publishPath = GetPublishFolder();
             if (string.IsNullOrEmpty(publishPath))
+            {
+                // Reset uploading state when user cancels folder selection
+                Uploading = false;
+                // Notify the front-end that the user has cancelled the upload
+                UploadCancelled?.Invoke();
```

Comment:
Notifying the front-end correctly switches off the publishing spinning wheel mode. We remain on the final screen afterwards ('Cancel' is not the same as 'Reset').